src/backend/distributed/utils/citus_depended_object.c

```diff
@@ -353,121 +344,21 @@ DistOpsHasInvalidObject(Node *node, const DistributeObjectOps *ops)
 		bool isPostprocess = false;
 		List *objectAddresses = ops->address(node, missingOk, isPostprocess);
 
+		bool hasAnyValidAddress = false;
 		ObjectAddress *objectAddress = NULL;
 		foreach_ptr(objectAddress, objectAddresses)
 		{
-			if (!OidIsValid(objectAddress->objectId))
+			if (OidIsValid(objectAddress->objectId))
 			{
-				return true;
+				hasAnyValidAddress = true;
```

Comment:
why don't we drop variable `hasAnyValidAddress` and return false immediately? 

We need more comments as well. Or, even a new ENUM could help understand this 3 state better
{
      HasAtLeastOneValidObject,
      HasNoneValidObject,
      NoAddressResolutionRequired
}

```
	if (ops && ops->address)
	{
		bool missingOk = true;
		bool isPostprocess = false;
		List *objectAddresses = ops->address(node, missingOk, isPostprocess);

		bool hasAnyValidAddress = false;
		ObjectAddress *objectAddress = NULL;
		foreach_ptr(objectAddress, objectAddresses)
		{
			if (OidIsValid(objectAddress->objectId))
			{
                   /* found one valid object */
                   return HasAtLeastOneValidObject;
			}
		}
         
         /* no valid objects */
		return HasNoneValidObject;
	}

    /* if the object doesn't have address defined, we cannot */
	return NoAddressResolutionRequired;
```

And,
```C
if (ops && ops->qualify && distOpsObj != HasNoneValidObject)
```

If we do so, the function name should also be changed to something like: